silq/instrument_interfaces/keysight/Keysight_81180A_interface.py

```diff
@@ -512,7 +515,7 @@ def add_pulse(self, pulse, t, channel, sample_rate):
             )
 
         # Get waveform of current pulse
-        waveform = pulse.implementation.implement(sample_rate=sample_rate,)
+        waveform = pulse.implementation.implement(sample_rate=sample_rate, pulse=pulse)
```

Comment:
This doesn't look right. Looking at all pulse implementations, it seems that DCPulseImplementation.implement is a staticmethod, while the rest isn't. So it looks like this line here would work for a DCPulse but not for other pulses. 

I think the correct solution is that DCPulseImplementation shouldn't have a staticmethod, though this might cause issues elsewhere

Reply:
Yes, the only reason I had to add this is that otherwise, it was giving me an error stating that it does not have the reference to which pulse it should use.

Reply:
I think the current change will not work for e.g. a SinePulse. I'll try to fix this when I have the time

Reply:
Yes, can confirm this. It worked for DCPulse, but for SinePulse gave an error.

Reply:
Try it now with the new commit I added